Code/Framework/AzFramework/Platform/Android/AzFramework/IO/LocalFileIO_Android.cpp

```diff
@@ -76,29 +77,22 @@ namespace AZ
         {
             ANDROID_IO_PROFILE_SECTION_ARGS("FindFiles:%s", filePath);
 
-            char resolvedPath[AZ::IO::MaxPathLength];
-            ResolvePath(filePath, resolvedPath, AZ::IO::MaxPathLength);
+            FixedMaxPath resolvedPath;
+            ResolvePath(resolvedPath, filePath);
+            resolvedPath = resolvedPath.LexicallyNormal();
 
-            AZStd::string pathWithoutSlash = RemoveTrailingSlash(resolvedPath);
-            bool isInAPK = AZ::Android::Utils::IsApkPath(pathWithoutSlash.c_str());
+            bool isInAPK = AZ::Android::Utils::IsApkPath(resolvedPath.c_str());
 
-            AZ::IO::FixedMaxPath tempBuffer;
             if (isInAPK)
             {
                 AZ::IO::FixedMaxPath strippedPath = AZ::Android::Utils::StripApkPrefix(pathWithoutSlash.c_str());
 
-                AZ::Android::APKFileHandler::ParseDirectory(strippedPath.c_str(), [&](const char* name)
+                AZ::Android::APKFileHandler::ParseDirectory(strippedPath.c_str(), [&](AZStd::string_view filenameView)
                     {
-                        AZStd::string_view filenameView = name;
                         // Skip over the current and parent directory paths
-                        if (filenameView != "." && filenameView != ".." && NameMatchesFilter(name, filter))
```

Comment:
why do we use AZStd::wildcard_match instead of old NameMatchesFilter?

Reply:
[NameMatchesFilter](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/FileIO.h#L31-L34) is a wrapper around `AZStd::wildcard_match` that reverses the parameters for no reason.
The function itself isn't needed and I am setting it up to be removed.